ethers-providers/src/lib.rs

```diff
@@ -283,6 +290,42 @@ pub trait Middleware: Sync + Send + Debug {
         self.inner().send_transaction(tx, block).await.map_err(FromErr::from)
     }
 
+    /// Send a transaction with a simple escalation policy.
+    ///
+    /// `escalation` should be a boxed function that maps `original_gas_price`
+    /// and `number_of_previous_escalations` -> `new_gas_price`.
+    ///
+    /// e.g. `Box::new(|start, escalations| start * 1250.pow(escalations) / 1000.pow(escalations))`
+    async fn send_escalating<'a>(
+        &'a self,
+        tx: &TypedTransaction,
+        from: &Address,
```

Comment:
You can use `self.default_sender` instead of `from`

Reply:
`default_sender()` returns `Option<H160>` what should be the behavior in the `None` case?

Reply:
I guess `unwrap_or_default`. that'll cause clear errors when we join_all later